src/test/java/firstGroup/Assignment1WithBase.java

```diff
@@ -0,0 +1,410 @@
+
+//********With base*************************
+
+    @Test
+    public void test1(){
+        System.out.println("Thread Id: " +  Thread.currentThread().getId());
+        getDriver().navigate().to("https://www.flipkart.com/");
+        startOperationBegin();
+        System.out.println("test1 = ");
+        Boolean b =   mainOperation();
+        Assert.assertTrue(b);
+    }
+    @Test
+    public void test2(){
+        System.out.println("Thread Id: " +  Thread.currentThread().getId());
+        getDriver().navigate().to("https://www.flipkart.com/");
+        startOperationBegin();
+        System.out.println("test2 = ");
+        Boolean b =   mainOperation();
+        Assert.assertTrue(b);
+    }
+    //***************Validate Add to Cart Functionality***************
+    @Test
+    public void test3(){
+        System.out.println("Thread Id: " +  Thread.currentThread().getId());
+        getDriver().navigate().to("https://www.flipkart.com/");
+        startOperationBegin();
+        System.out.println("Done startOperationBegin");
+        addProduct();
+
+    }
+
+    public void addProduct(){
+
+        Assignment1Definition obj1 = new Assignment1Definition(driver.get());
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        String shoeOneTextFromList = obj1.getTextOfShoeFirst().getText();
+        System.out.println("shoeOneTextFromList = "+shoeOneTextFromList);
+        obj1.clickFirstShoe().click();
+
+//        Set<String> windows =	driver.getWindowHandles();
+//Thread.sleep
+
+        Set<String> windows =   driver.get().getWindowHandles();
+
+        Iterator<String> itrWindows = windows.iterator();
+        String parentWindow=itrWindows.next();
+        String child1Window=itrWindows.next();
+
+
+
+        driver.get().switchTo().window(child1Window);
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        obj1.selectShoeSize().click();
+        try {
+            Thread.sleep(3000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        obj1.selectAddToCart().click();
+
+        System.out.println("Done");
+
+
+////***************switch to parent window***************
+
+        driver.get().switchTo().window(parentWindow);
+
+        try {
+            Thread.sleep(2000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+//***********Selecting 2nd shoe*******************
+        String shoeTowTextFromList = obj1.getTextOfShoeSecond().getText();
+        System.out.println("shoeTowTextFromList = "+shoeTowTextFromList);
+        obj1.clickSecondShoe().click();
+
+//***********switching to child window***********
+        Set<String> windows2 =   driver.get().getWindowHandles();
+        Iterator<String> itrWindows2 = windows2.iterator();
```

Comment:
instead of defining a Set windows, use :
Iterator itrWindows2 = driver.get().getWindowHandles().iterator();